controller/src/main/java/io/pravega/controller/task/KeyValueTable/TableMetadataTasks.java

```diff
@@ -0,0 +1,271 @@
+                                                        .thenCompose(uuid -> {
+                                                            CreateTableEvent event = new CreateTableEvent(scope, kvtName, kvtConfig.getPartitionCount(),
+                                                                                                        createTimestamp, requestId, UUID.fromString(uuid));
+                                                            return addIndexAndSubmitTask(event,
+                                                                    () -> CompletableFuture.completedFuture(null))
+                                                                    .thenCompose(x -> checkDone(() -> isCreated(scope, kvtName, kvtConfig, executor))
+                                                                            .thenCompose(y -> kvtMetadataStore.getConfiguration(scope, kvtName, null, executor)
+                                                                            .thenCompose(cfg -> {
+                                                                                if (cfg.getPartitionCount() == kvtConfig.getPartitionCount()) {
+                                                                                   return CompletableFuture.completedFuture(CreateKeyValueTableStatus.Status.SUCCESS);
+                                                                                } else {
+                                                                                   return CompletableFuture.completedFuture(CreateKeyValueTableStatus.Status.TABLE_EXISTS);
+                                                                                }
+                                                                            })));
+                                                        })),
+                                                e -> Exceptions.unwrap(e) instanceof StoreException.DataExistsException,
+                                                CreateKeyValueTableStatus.Status.TABLE_EXISTS);
+                                    } else {
+                                        return CompletableFuture.completedFuture(CreateKeyValueTableStatus.Status.SCOPE_NOT_FOUND);
+                                    }
+                                }), executor);
+                    } else {
+                       return CompletableFuture.completedFuture(CreateKeyValueTableStatus.Status.TABLE_EXISTS);
+                    }
+                }), e -> Exceptions.unwrap(e) instanceof RetryableException, Integer.MAX_VALUE, executor);
+    }
+
+    private CompletableFuture<Void> checkDone(Supplier<CompletableFuture<Boolean>> condition) {
+        return checkDone(condition, 100L);
+    }
+
+    private CompletableFuture<Void> checkDone(Supplier<CompletableFuture<Boolean>> condition, long delay) {
+        AtomicBoolean isDone = new AtomicBoolean(false);
```

Comment:
please pull latest master and use the timeout mechanism used there..

Reply:
The upstream for this branch, is feature branch `feature-key-value-tables` on `pravega/pravega` and that is not updated yet with changes from master. I'd want to put this off till the upstream is updated..... Can add this as part of my next PR for DeleteAPI.

Reply:
Sure. Add a todo comment here please.